site/src/components/BackToTopButton.tsx

```diff
@@ -0,0 +1,61 @@
+import {useState, useEffect} from 'react';
+
+
+export default function BackToTopButton(){
+	const[isVisible, setIsVisible] = useState(false);
+
+	// show button when page scrolldown
+	const toggleVisibility =() =>{
+		if(window.pageYOffset > 300){
+			setIsVisible(true);
+		}else{
+			setIsVisible(false);
+		}
+	};
+
+
+	// Scroll to top smoothly
+	const scrollToTop = () => {
+		window.scrollTo({
+			top:0,
+			behavior: 'smooth',
+		});
+	};
+
+
+
+	useEffect(()=>{
+		window.addEventListener('scroll', toggleVisibility);
+		return()=>{
+			window.removeEventListener('scroll', toggleVisibility);
+		};
+	}, []);
+
+
+	if(!isVisible){
+		return null;
+	}
+
+
```

Comment:
Your whitespace is _super weird_ here? `=() =>{` for `toggleVisibility`, but `= () => {` for `scrollToTop`, and multiple newlines randomly between blocks of logic?

Can you, as a human, not AI, please go through and apply a much more consistent approach to your whitespace.